clang/lib/Driver/Driver.cpp

```diff
@@ -8457,9 +8457,17 @@ class ToolSelector final {
     return dyn_cast<JobAction>(CurAction);
   }
 
+  // We need to collapse the separate compilation steps when performing
+  // a third party host compilation step for SYCL offloading.  We don't know
+  // what the third party compiler is capable of, so only allow for object
+  // creation when performing -save-temps.
+  bool SYCLHostCompiler =
+      BaseAction->isHostOffloading(Action::OFK_SYCL) &&
+      C.getArgs().hasArg(options::OPT_fsycl_host_compiler_EQ);
+
   /// Return true if an assemble action can be collapsed.
```

Comment:
nit: This comment can be updated to include the new return condition.

Reply:
I think these comments can stay as-is.  They are generic comments on what the function does and doesn't pinpoint specifics.  There are a number of different existing conditions for each that can be met to prevent the collapsing and these are not called out either.

Reply:
>>The scenarios when we want to collapse:

> - default compilation && !save-temps
> - fsycl-host-compiler && save-temps (host compilation steps only)

Based on your comment above, my understanding is the function combines the compilation only for the conditions listed above, which is why I suggested to update the comment.